Scripts/assignment/emme_assignment.py

```diff
@@ -724,46 +673,178 @@ def _calc_extra_wait_time(self, scen_id):
+        # save matrix results for both classes
+        for name, spec in zip(transit_classes, tcs):
+            self.emme_project.matrix_results(spec.transit_result_spec, scenario=scen, class_name=name)
+            self.emme_project.network_results(spec.ntw_results_spec, scenario=scen, class_name=name)
+        self.emme_project.logger.info(
+            "Congested transit assignment performed for scenario {}".format(str(scen_id)))
+
+    def auto_link_24h(self, attr):
+        """ 
+        Sums and expands link volumes from different scenarios to one result scenario.
+         Parameters
+        ----------
+        attr : str
+            Attribute name thatis usually part of Parameters: link_volumes.
+        """
+        extra_attr = param.link_volumes[attr]
+        # get attr from different time periods to dictionary
+        links_attr = {}
+        for tp in self.emme_scenarios:
+            tp_attr = {}
+            emmebank = self.emme_project.modeller.emmebank
+            scenario = emmebank.scenario(self.emme_scenarios[tp])
+            network = scenario.get_network()
+            for link in network.links():
+                tp_attr[link.id] = link[extra_attr]
+            links_attr[tp] = tp_attr
+        # create attr to save volume
+        extra_attr_day = str(param.link_volumes[attr])
+        day_scenario = emmebank.scenario(self.day_scenario)
+        network = day_scenario.get_network()
+        # save link volumes to result network
+        for link in network.links():
+            day_attr = 0
+            for tp in self.emme_scenarios:
+                if link.id in links_attr[tp]:
+                    day_attr += links_attr[tp][link.id] * param.volume_factors[attr][tp]
+            link[extra_attr_day] = day_attr
+        day_scenario.publish_network(network)
+
+    def transit_segment_volumes_24h(self, attr):
+        """ 
+        Sums and expands transit volumes to 24h.
+        """
+        extra_attr = "@{}_vol".format(attr)
+        # get attr from different time periods to dictionary
+        segments_attr = {}
+        for tp in self.emme_scenarios:
+            tp_attr = {}
+            emmebank = self.emme_project.modeller.emmebank
+            scenario = emmebank.scenario(self.emme_scenarios[tp])
+            network = scenario.get_network()
+            for segment in network.transit_segments():
+                tp_attr[segment.id] = segment[extra_attr]
+            segments_attr[tp] = tp_attr
+        # save link volumes to result network
+        day_scenario = emmebank.scenario(self.day_scenario)
+        network = day_scenario.get_network()
+        for segment in network.transit_segments():
+            day_attr = 0
+            for tp in self.emme_scenarios:
+                if segment.id in segments_attr[tp]:
+                    day_attr += segments_attr[tp][segment.id] * param.volume_factors["transit"][tp]
```

Comment:
Need to use the class-specific volume factor

Reply:
https://github.com/HSLdevcom/helmet-model-system/pull/183/commits/70c6f395516fb2ed37dcf23108fee46d36f0525d